content/blog/2018-05-23-react-v-16-4.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "React v16.4.0: Pointer Events"
+author: [acdlite]
+---
+
+The latest minor release adds support for an oft-requested feature: pointer events!
+
+It also includes a bugfix for `getDerivedStateFromProps`. Check out the full [changelog](#changelog) below.
+
+## Pointer Events
+
+The following event types are now available in React DOM:
+
+- `onPointerDown`
+- `onPointerMove`
+- `onPointerUp`
+- `onPointerCancel`
+- `onGotPointerCapture`
+- `onLostPointerCapture`
+- `onPointerEnter`
+- `onPointerLeave`
+- `onPointerOver`
+- `onPointerOut`
+
+Please note that these events will only work in browsers that support the Pointer Events specification. (At the time of this writing, this includes the latest versions of Chrome, Firefox, Edge, and Internet Explorer.) If your application depends on pointer events, we recommend using a third-party pointer events polyfill. We have opted not to include such a polyfill in React DOM, to avoid an increase in bundle size.
+
+[Check out this example on CodeSandbox.](codesandbox://16-4-release-blog-post/pointer-events-example)
+
+Huge thanks to [Philip Spiess](https://github.com/philipp-spiess) for contributing this change!
+
+## Bugfix for `getDerivedStateFromProps`
+
+`getDerivedStateFromProps` is now called every time a component is rendered, regardless of the cause of the update. Previously, it was only called if the component was re-rendered by its parent, and would not fire as the result of a local `setState`. This was an oversight in the initial implementation that has now been corrected. The previous behavior was more similar to `componentWillReceiveProps`, but the improved behavior ensures compatibility with React's upcoming asynchronous rendering mode.
+
+**This bug fix will not affect most apps**, but it may cause issues with a small fraction of components. The rare cases where it does matter fall into one of two categories:
+
+### 1. Avoid Side Effects in `getDerivedStateFromProps`
+
+Like the render method, `getDerivedStateFromProps` should be a pure function of props and state. You should move side-effectful code elsewhere. E.g. Flux dispatches typically belong inside the originating event handler. DOM mutations belong inside `componentDidMount` or `componentDidUpdate`.
+
+Side effects in `getDerivedStateFromProps` were already disallowed, but since it now fires more often than it used to, the recent change may expose previously undiscovered bugs.
+
+You can read more about this in our recent post about [preparing for asynchronous rendering](/blog/2018/03/27/update-on-async-rendering.html).
+
+### 2. Compare Incoming Props to Previous Props When Computing Controlled Values
+
+The following code assumes `getDerivedStateFromProps` only fires on prop changes:
+
+```js
+static getDerivedStateFromProps(props, state) {
+  if (props.value !== state.controlledValue) {
+    return {
+      // Since this method fires on both props and state changes, local updates
```

Comment:
Super tiny nit. Please feel free to ignore.

Suggestion: Change "_Since this method fires_" to "_Because this method fires_"